src/Compiler/pars.fsy

```diff
@@ -4107,6 +4107,22 @@ moreBinders:
        let trivia = { AndBangKeyword = rhs parseState 1; EqualsRange = mEquals; InKeyword = Some mIn }
        SynExprAndBang(spBind, $1, true, $2, $4, m, trivia) :: $6 }
 
+  | AND_BANG headBindingPattern opt_topReturnTypeWithTypeConstraints EQUALS typedSequentialExprBlock IN moreBinders %prec expr_let
```

Comment:
At this point we should probably try to unify `let` and `let!` parsing rules and check whether it's a `!` version or not in the type checker. As the first step it should be at least parse it in a unified manner and then create different tree nodes conditionally. But ideally it should be something like `isComputed: bool` flag that the type checker would inspect.

@edgarfgp Do you think it's feasible to unify the rules as part of this PR?

Reply:
> Do you think it's feasible to unify the rules as part of this PR?

Agreed. I think as part of this PR I can try and I can try and unify the parsing rules for `let`, `let!` and `and!`.  

> Could you also cover error cases in the grammar, please?

Yeah I can add syntax tree test that show the parsing errors.

cc @T-Gro any objections ?

Reply:
No objections.
If we could reduce the number of related rules effectively in half, and then possibly cover the difference in typechecker via active patterns to split it between `let` and `let!`  (to minimize code differences)